### DISCUSSION TITLE
Update error 404

### DISCUSSION THREAD

**Discussion by Tihomir971** [supabase/supabase#5830] [Questions] [answered]
With
const { data, error } = await supabase.from('product').update(productRow).eq('sku', sku);
i get error in console
PATCH https://xxx.supabase.co/rest/v1/product?sku=eq.0000000000123 404

Problem is that sku in table doesn't exist but I can't handle error. Error is empty object and there is always error in console.

I try with upsert but with same results. 
I didn't set RLS

Any suggestions?

**Comment by GaryAustin1** [comment]
The 404 comes from the server if row is not found, obviously.
Supabase.js does not do anything with the 404 as far as trying to map it to error.
You will get an empty array in data if no rows were found/modified (unless you turn on option returning:'minimal') if you want to see if modified.
.upsert should insert the row if it does not exist.

**Reply by Tihomir971** [reply]
It means supabase error is useless :)
I can't use upsert because I dont have primary id (I'm importing products from csv)
Is the only solution to first select  then insert or update?

**Comment by GaryAustin1** [comment, marked as answer]
If you make your sku column have property unique then you can tell upsert to look at that also in deciding.  https://supabase.com/docs/reference/javascript/upsert#upserting-into-tables-with-constraints.
